lib/resty/rediscluster.lua

```diff
@@ -29,17 +29,26 @@ local DEFAULT_CONNECTION_TIMEOUT = 1000
 local DEFAULT_SEND_TIMEOUT = 1000
 local DEFAULT_READ_TIMEOUT = 1000
 local DEFAULT_HEALTH_DICT_NAME = "redis_cluster_health"
-local inspect = require("inspect")
 local err_unhealthy_master = "master node is unhealthy"
+
+local function generate_key(name, ip, port)
+    return name .. ":" .. ip .. ":" .. port
+end
+
+
 local function health_check_timer(premature)
-    if premature then return end
+    if premature then
+        return
+    end
 
     local health_dict = ngx.shared[DEFAULT_HEALTH_DICT_NAME]
-    if not health_dict then return end
+    if not health_dict then
+        return
+    end
 
     local all_keys = health_dict:get_keys()
     for _, key in ipairs(all_keys) do
-        local ip, port = string.match(key, "^(.-):(%d+)$")
+        local ip, port = string.match(key, "^[^:]+:([^:]+):(%d+)$")
```

Comment:
An optimization suggestion: In multiple Redis cluster configurations, the same IP port may be included. You can first deduplicate to avoid repeatedly checking the same instance.

Reply:
I found we can achive this optimization by reuse same cluster name in multiple redis cluster configuration, I think it's enough.